src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -408,9 +408,47 @@ private class Emit(
           xvout := xmout.mux(defaultValue(typ), xvout)
         ), xmout, xvout)
 
-      case x@ApplyAggOp(a, op, args) =>
-        val agg = AggOp.get(op, x.inputType, args.map(_.typ))
-        present(emitAgg(a)(agg.seqOp(aggregator, _, _)))
+      case ArrayFor(a, valueName, body) =>
+        val tarray = coerce[TArray](a.typ)
+        val eti = typeToTypeInfo(tarray.elementType)
+        val xmv = mb.newField[Boolean]()
```

Comment:
I realize we're using `newField[Boolean]` elsewhere already, but we could use `newClassBit` to get a bit-packed thing. The trade-offs are unclear to me. obviously those 64 bits will often stay in the cache, but a handful of ops are necessary to mask.

Reply:
So @catoverdrive had a chat about this, and I'm not sure bits are preferred.  It greatly increases the size of the bytecode and introduces lots of false dependencies.  I think we should benchmark, but I wonder if `newField[Boolean]` is going to actually be better.